cmd/podman/shared/container.go

```diff
@@ -609,7 +609,7 @@ func GetRunlabel(label string, runlabelImage string, ctx context.Context, runtim
 			registryCreds = creds
 		}
 		dockerRegistryOptions.DockerRegistryCreds = registryCreds
-		newImage, err = runtime.ImageRuntime().New(ctx, runlabelImage, signaturePolicyPath, authfile, output, &dockerRegistryOptions, image.SigningOptions{}, false)
+		newImage, err = runtime.ImageRuntime().New(ctx, runlabelImage, signaturePolicyPath, authfile, output, &dockerRegistryOptions, image.SigningOptions{}, false, &label)
```

Comment:
Hum… so, ultimately, this code does not need the image layers pulled _at all_, and all it only needs the labels? That’s fairly wasteful (though it’s not easy to get the `pullGoal` code in any other way, right now).

Or would the image get pulled either way?

Reply:
> if the label doesn't exist in the remote image, we don't pull it because the image would be useless.

Sure, if the label does not exist; but if the label _does_ exist, will the image get pulled soon after, or is that possible to be a big waste?

Reply:
Per IRC, the image would get pulled if the label exists.